src/main/kotlin/com/zopa/ktor/opentracing/OpenTracingServer.kt

```diff
@@ -82,6 +84,16 @@ class OpenTracingServer {
                 }
             }
 
+            pipeline.environment.monitor.subscribe(Routing.RoutingCallStarted) {call ->
+                if (config.filters.any { it(call) }) return@subscribe
+                val span = GlobalTracer.get().activeSpan() ?: return@subscribe
+
+                call.parameters.entries().forEach { param ->
+                    span?.setTag(param.key, param.value.first())
+                }
+                span.setOperationName("${call.request.httpMethod.value} ${call.route.parent}")
```

Comment:
the parent field is nullable, so if it's null we wipe the span name, maybe we could do this in a 
```kotlin
call.route.parent?.let {
```

Reply:
also, why are you using the parent field of `call.route` rather than `call.route`?

Reply:
if we don't have a good understanding of how that works , it's probably more reasonable to construct this string by replacing the param key / values in the `call.request.path()`

Reply:
The way it works is that ktor creates a tree of routes - every section of the path is another child, down to the last child which is the httpMethod. So although parent CAN be null, it's only in the case of the root route that it will be, and we won't have matched to that with the call.
I'm only using the `parent` to match the formatting you have used in the other intercept. `call.route.toString()` would have all the information, but we already have "METHOD PATH" as the format. (I've forgotten exactly what the format would be, something like "PATH (METHOD)".)